src/services/invitation/service.ts

```diff
@@ -136,4 +140,98 @@ export class InvitationService {
+
+    if (!file || file.mimetype != CSV_MIMETYPE) {
+      throw new Error(`
+          An incorrect type of file has been uploaded, 
+          Please upload a file with .csv extension
+        `);
+    }
+
+    const stream = file.file;
+    const { rows, header } = await getCSV(stream);
+    const hasGrpCol = header.includes(GRP_COL_NAME);
+    let groupNameToItem = new Map<string, Item>();
+
+    if (hasGrpCol) {
+      groupNameToItem = await rows.reduce(
+        async (result: Promise<Map<string, Item>>, rowInvite: CSVInvite) => {
+          const r = await result;
+          if (!r.has(rowInvite.group_name as string)) {
+            const itemCreated = await this.itemService.post(actor, repositories, {
+              item: {
+                name: rowInvite.group_name,
+                type: ItemType.FOLDER,
+              },
+              parentId: parentId,
+            });
+
+            if (template_id) {
+              //get the all items of from template_id and copy them to new folder
+              const itemsInTemplate = await this.itemService.getDescendants(
+                actor,
+                repositories,
+                template_id,
+              );
+              const regex = regexGen(
+                (await this.itemService.get(actor, repositories, template_id)).path,
+              );
+
+              const itemsAtFirstLevel = itemsInTemplate
+                .filter((tmpItem) => {
+                  return regex.test(tmpItem.path);
+                })
+                .map((tmpItem) => {
+                  return tmpItem.id;
+                });
+
+              await this.itemService.copyMany(actor, repositories, itemsAtFirstLevel, {
+                parentId: itemCreated.id,
+              });
+            }
+            r.set(rowInvite.group_name as string, itemCreated);
+          }
+          return r;
+        },
+        Promise.resolve(new Map<string, Item>()),
+      );
+    }
+
+    const invitesPerGroup = rows.reduce((result, rowInvite: CSVInvite) => {
+      let group_name = '';
+      if (hasGrpCol) group_name = rowInvite.group_name as string;
+      const invite = new Invitation();
+      invite.email = rowInvite['email'];
+      invite.permission = PermissionLevel.Read;
+      if (rowInvite.permission) invite.permission = rowInvite.permission;
+      if (!rowInvite.name) invite.name = '';
+      if (hasGrpCol) invite.item = groupNameToItem.get(group_name) as Item;
+      else invite.item = parentItem;
+      if (!result.has(group_name)) {
+        result.set(group_name, [invite]);
+      } else {
+        result.get(group_name)?.push(invite);
+      }
+      return result;
+    }, new Map<string, Invitation[]>());
+
+    for await (const arrOfInvitations of invitesPerGroup.values()) {
+      await this.postManyForItem(
+        actor,
+        repositories,
+        arrOfInvitations[0].item.id,
+        arrOfInvitations,
+      );
+    }
```

Comment:
You can use a `await Promise.all(invitesPerGroup.values().map( async (i) => this.postManyForItem(...)))` instead.